flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JobExceptionsHandler.java

```diff
@@ -41,19 +43,23 @@ public JobExceptionsHandler(ExecutionGraphHolder executionGraphHolder) {
 
 	@Override
 	public String handleRequest(AccessExecutionGraph graph, Map<String, String> params) throws Exception {
+		return createJobExceptionsJson(graph);
+	}
+
+	public static String createJobExceptionsJson(AccessExecutionGraph graph) throws IOException {
 		StringWriter writer = new StringWriter();
 		JsonGenerator gen = JsonFactory.jacksonFactory.createGenerator(writer);
 
 		gen.writeStartObject();
 		
 		// most important is the root failure cause
 		String rootException = graph.getFailureCauseAsString();
-		if (rootException != null) {
-			gen.writeStringField("root-exception", rootException);
+		if (!rootException.equals(ExceptionUtils.STRINGIFIED_NULL_EXCEPTION)) {
```

Comment:
Then maybe add a nonNull check to guard against unintended future changes of behaviour in ExecutionGraph?